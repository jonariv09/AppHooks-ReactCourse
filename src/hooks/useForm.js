import { useState } from 'react'

export const useForm = (target = {}) => {
  
  const [formState, setFormState] = useState(target);

  const handleInputChange = ({ target }) => {
    setFormState({
      ...formState,
      [target.name]: target.value
    })
  }

  return {
    formState,
    handleInputChange
  }

}
