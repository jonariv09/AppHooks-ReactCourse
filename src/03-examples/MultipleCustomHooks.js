import React from 'react'
import { useCounter } from '../hooks/useCounter';
import { useFetch } from '../hooks/useFetch';

export const MultipleCustomHooks = () => {

  const { counter, increment } = useCounter(1);
  const { loading, data } = useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`);

  const { quote, author } = !!data && data[0];


  return (
    <div>
      <h1> Breaking Bad </h1>
      <hr />
      
      {
        loading
        ? 
          (
            <div className="loader"> Loading.... </div>
          )
        :
          (
            <blockquote>
              <p className="quote"> { quote } </p>
              <footer className="author"> { author }  </footer>
            </blockquote>
          )
      }

      <button onClick={increment}> Siguiente quote </button>

    </div>
  )
}
