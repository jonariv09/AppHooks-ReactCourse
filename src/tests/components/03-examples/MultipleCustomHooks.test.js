import {renderHook, act} from '@testing-library/react-hooks';
import { shallow } from 'enzyme';
import { MultipleCustomHooks } from '../../../03-examples/MultipleCustomHooks';
import { useFetch } from '../../../hooks/useFetch';

jest.mock('../../../hooks/useFetch');

describe('Pruebas en <MultipleCustomHooks />', () => {

  test('debe mostrarse correctamente', () => {

    useFetch.mockReturnValue({
      data: null,
      loading: true,
      error: null
    });

    const wrapper = shallow(<MultipleCustomHooks />);
    expect(wrapper).toMatchSnapshot();

  })

  test('debe de mostrar la informacion', () => {

    useFetch.mockReturnValue({
      data: [{
        author: 'Agustin',
        quote: 'Hola Mundo'
      }],
      loading: false,
      error: null
    });

    const wrapper = shallow(<MultipleCustomHooks />);

    expect(wrapper.find('.loader').exists()).toBe(false);
    expect(wrapper.find('.quote').text().trim()).toBe('Hola Mundo');
    expect(wrapper.find('.author').text().trim()).toBe('Agustin');
  })

})