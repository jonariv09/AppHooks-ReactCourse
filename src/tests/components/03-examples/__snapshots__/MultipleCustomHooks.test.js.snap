// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomHooks /> debe mostrarse correctamente 1`] = `
<div>
  <h1>
     Breaking Bad 
  </h1>
  <hr />
  <div
    className="loader"
  >
     Loading.... 
  </div>
  <button
    onClick={[Function]}
  >
     Siguiente quote 
  </button>
</div>
`;
